backend/src/services/club.go

```diff
@@ -87,5 +91,63 @@ func (c *ClubService) DeleteClub(id string) *errors.Error {
 		return &errors.FailedToValidateID
 	}
 
-	return transactions.DeleteClub(c.DB, *idAsUUID)
+return transactions.DeleteClub(c.DB, *idAsUUID)
+}
+
+func (c *ClubService) GetContacts(limit string, page string) ([]models.Contact, *errors.Error) {
+	limitAsInt, err := utilities.ValidateNonNegative(limit)
+
+	if err != nil {
+		return nil, &errors.FailedToValidateLimit
+	}
+
+	pageAsInt, err := utilities.ValidateNonNegative(page)
+
+	if err != nil {
+		return nil, &errors.FailedToValidatePage
+	}
+
+	offset := (*pageAsInt - 1) * *limitAsInt
+
+	return transactions.GetContacts(c.DB, *limitAsInt, offset)
+}
+
+func (c *ClubService) GetClubContacts(id string) ([]models.Contact, *errors.Error) {
+	idAsUUID, err := utilities.ValidateID(id)
+	if err != nil {
+		return nil, &errors.FailedToValidateID
+	}
+
+	return transactions.GetClubContacts(c.DB, *idAsUUID)
+}
+
+func (c *ClubService) PutContact(clubID string, contactBody models.PutContactRequestBody) (*models.Contact, *errors.Error) {
+	idAsUUID, idErr := utilities.ValidateID(clubID)
+	if idErr != nil {
+		return nil, idErr
+	}
+
+	if err := c.Validate.Struct(contactBody); err != nil {
+		return nil, &errors.FailedToValidateContact
+	}
+
+	contact, err := utilities.MapRequestToModel(contactBody, &models.Contact{})
+	if err != nil {
+		return nil, &errors.FailedToMapRequestToModel
+	}
+
+	if *idAsUUID != contact.ClubID {
```

Comment:
I don't know if we need this check. The request body probably shouldn't take an ID, rather source it from the params